dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/ContainerClassLoader.java

```diff
@@ -108,6 +107,69 @@ public static void disableJarCaching() {
         }
     }
 
+    static final Set<Name> packageAttributes;
+
+    static {
+        Set<Name> packageAttrs = new HashSet<>();
+        packageAttrs.add(Name.SPECIFICATION_TITLE);
+        packageAttrs.add(Name.SPECIFICATION_VERSION);
+        packageAttrs.add(Name.SPECIFICATION_VENDOR);
+        packageAttrs.add(Name.IMPLEMENTATION_TITLE);
+        packageAttrs.add(Name.IMPLEMENTATION_VERSION);
+        packageAttrs.add(Name.IMPLEMENTATION_VENDOR);
+        packageAttrs.add(Name.SEALED);
+        packageAttributes = Collections.unmodifiableSet(packageAttrs);
+    }
+
+    /**
+     * Load manifest file and filter out attributes that are not needed by definePackage.
+     * 
+     * Since the manifest will be kept in memory only keep that which is required by this class. 
+     * 
+     * @param manifestStream InputStream to read the manifest file
+     * @return filtered Manifest 
+     * @throws IOException
+     */
+    static Manifest loadManifest(InputStream manifestStream) throws IOException {
```

Comment:
Is there any way to cache the results of this?  At least to cache if there are any headers we care about (true/false) so we can avoid parsing the manifest altogether for JARs that do not have these headers?  Manifest parsing can be expensive for signed JAR's in particular because they have so many more entries to parse the signer information for.

Reply:
We probably could.  Sounds like a good next step in this startup journey.